ui/jobs/components/ast.ts

```diff
@@ -97,9 +97,7 @@ export class ASTComponent extends BaseComponent {
     // Show whether you already have this seal
     // ○ means it's OK to play this card
     // × means you'd better redraw if possible
-    if (!cardsMap[card])
-      this.cardBox.innerText = '';
-    else if (sign.includes(cardsMap[card].seal))
+    if (sign.includes(cardsMap[card].seal))
```

Comment:
Cannot remove O/X after play the card. Sign color also not updated until next draw.
With the following error:
```
TypeError: Cannot read properties of undefined (reading 'seal')
    at ASTComponent.onJobDetailUpdate (ast.ts:100)
```

Reply:
The typing of `cardsMap` doesn't allow `undefined`, this seems like a bug in general.

```ts
const cardsMap = {
  'Balance': { 'bonus': 'melee', 'seal': 'Solar' },
  'Bole': { 'bonus': 'range', 'seal': 'Solar' },
  'Arrow': { 'bonus': 'melee', 'seal': 'Lunar' },
  'Ewer': { 'bonus': 'range', 'seal': 'Lunar' },
  'Spear': { 'bonus': 'melee', 'seal': 'Celestial' },
  'Spire': { 'bonus': 'range', 'seal': 'Celestial' },
} as const;
```

If this change is breaking functionality then something related to either `cardsMap` or `card` is not defined correctly and should be updated. I'm guessing maybe `JobDetail['AST'].heldCard` should be optional or allow null?

Reply:
This is not a matter of `cardsMap`. If player do not held a card, card will be `'None'`, and `cardsMap[card]` returns `undefined`. If plugin remains unchanged then there should be a condition.
By the way, I think `jobDetail.heldCard` and `jobDeail.crownCard` should return `null/undefined` when no card.

Reply:
They should map to `'None'` when no card is held, if I'm reading the corresponding C# code properly? So the proper typedef would be:
```ts
  'AST': {
    heldCard: 'None' | 'Balance' | 'Bole' | 'Arrow' | 'Spear' | 'Ewer' | 'Spire';
    crownCard: 'None' | 'Lord' | 'Lady';
    arcanums: ('Solar' | 'Lunar' | 'Celestial')[];
  };
```

Corresponding C# code just toString's the enum:
```cs
      public string heldCard {
        get {
          return ((Card)(_heldCard & 0xF)).ToString();
        }
      }

      public string crownCard {
        get {
          return ((Card)(_heldCard & 0xF0)).ToString();
        }
      }
```

Reply:
Seems both problem solved. Thank you!